clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/gitRepo/GitRepoArtifactCredentials.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2019 Armory
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.artifacts.gitRepo;
+
+import com.netflix.spinnaker.clouddriver.artifacts.config.ArtifactCredentials;
+import com.netflix.spinnaker.kork.artifacts.model.Artifact;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.jgit.api.ArchiveCommand;
+import org.eclipse.jgit.api.CheckoutCommand;
+import org.eclipse.jgit.api.CloneCommand;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.archive.ArchiveFormats;
+import org.eclipse.jgit.transport.UsernamePasswordCredentialsProvider;
+
+@Slf4j
+public class GitRepoArtifactCredentials implements ArtifactCredentials {
+  @Getter private final List<String> types = Collections.singletonList("git/repo");
+
+  @Getter private final String name;
+  private final String username;
+  private final String password;
+
+  public GitRepoArtifactCredentials(GitRepoArtifactAccount account) {
+    this.name = account.getName();
+    this.username = account.getUsername();
+    this.password = account.getPassword();
+  }
+
+  @Override
+  public InputStream download(Artifact artifact) throws IOException {
+    String repoReference = artifact.getReference();
+    Path stagingPath =
+        Paths.get(System.getProperty("java.io.tmpdir"), UUID.randomUUID().toString());
+
+    try {
+      log.info("Cloning git/repo {} into {}", repoReference, stagingPath.toString());
+      Git localRepository = clone(artifact, stagingPath);
+      ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+      log.info("Creating archive for git/repo {}", repoReference);
+      archiveToOutputStream(artifact, localRepository, outputStream);
+      return new ByteArrayInputStream(outputStream.toByteArray());
+    } catch (Exception e) {
```

Comment:
It would be better to catch a more specific exception here (which may be possible in light of the comment below on declaring something more specific in the `throws` clauses).